packages/contract/src/contracts/contract.ts

```diff
@@ -62,4 +63,15 @@ export default class Contract implements AbstractContract {
   multiCall(calls: Array<FunctionInvocationScope>) {
     return new MultiCallInvocationScope(this, calls);
   }
+
+  /**
+   * Get the balance for a given assset ID for this contract
+   */
+  getBalance(assetId: BytesLike) {
+    if (!this.provider) {
```

Comment:
Do you know if there's a case where a contract can be used without a provider?

Reply:
Not sure if it can be _used_ without a provider but I think it can technically be _initialized_ if we don't pass in a provider + we pass in a wallet that does not contain a `provider`. See [this code](https://github.com/FuelLabs/fuels-ts/blob/master/packages/contract/src/contracts/contract.ts#L44-L47)

Reply:
I think it is not possible to not have a provider in this case, as the constructor assigned the provider from the wallet or the provider it self.

Reply:
> **Note** Just to make it clear, this point I'm touching on might be out of the scope of this PR, but if we agree on changing the params signature, it could become a new issue.

Reply:
@luizstacio thoughts on this? I can file a separate issue for this if we want to make this change, and we can go ahead with this PR as it is

Reply:
We should remove the null, as the Contract will not work without a provider or wallet.